tutorials/haskell/hspec-golden-test-tutorial/tutorial.md

```diff
@@ -0,0 +1,565 @@
+
+For example, lets create a function that outputs the string "Hello Golden Testers":
+```haskell
+sayHi :: String
+sayHi = "Hello Golden Testers"
+
+```
+
+Using Hspec we can easily test our function :
+
+```haskell
+import HelloWorld
+import Test.Hspec
+
+spec :: Spec
+spec = do
+    describe "sayHi" $
+        it "shows a Hello Golden Testers string" $
+            sayHi `shouldBe` "Hello Golden Testers"
+```
+
+So, in a nutshell we are comparing the output of the function with a string that is stored inside the code of our test. We can take a different approach and store this expected output in a separate file. This approach is known as golden testing and the file in which we store the expected output takes the name of "golden file".
+
+So, now we know what golden testing is, but if unit tests and golden tests are so similar, why don't we just keep up with unit tests, right?
+
+The truth is that unit tests are not useful when evaluating complex and large outputs and that's where the difference lies. Here are some advantages of golden tests over unit tests.
+
+ - It wouldn't be practical to store a large output inside the test code.
+
+ - With some golden tests libraries the expected output (Golden File) can be automatically generated.
+
+ - Some golden test libraries also automatically updates the golden file. So, we don't need to manually update the tests when the expected output changes. 
+
+ - It is useful when working with JSON, HTML, images, etc.
+
+In this tutorial we will use [hspec-golden](http://hackage.haskell.org/package/hspec-golden) library to build some golden-test. We will also use the hspec-golden CLI to easily update our tests.
+
+You can either code along with this tutorial or better undestand it by looking at the finished code. You can find this [code](https://github.com/stackbuilders/tutorials/tree/jguerra-hspec-tutorial) in Github. Feel free to use it.
+
+## Hspec-golden
+
+Hspec-golden is a stackbuilders testing module written in haskell that helps users implement golden test. It is an OpenSource project so feel free to [review](https://github.com/stackbuilders/hspec-golden) it.
+
+## Getting Started
+
+First things first, lets create a new Haskell project on which we will be working:
+
+```bash
+stack new "hspec-golden-tests"
+```
+
+and add the necessary dependencies in order to follow this tutorial without issues.
+
+Project dependencies
+
+``` yaml
+dependencies:
+- base >= 4.7 && < 5
+- aeson
+- text
+- bytestring
+- blaze-html
+```
+
+Testing dependencies
+
+``` yaml
+
+dependencies:
+- hspec-golden-tests
+- hspec
+- hspec-golden
+
+```
+## A Simple Test
+
+Now that we have all dependencies installed, lets create our golden tests.
+
+First lets create a ```FizBuz ``` module to be tested. For convinience I have created a FIZBUZ directory. Now let's add some code to our module. 
```

Comment:
@Jagl257 I think the correct name is `FizzBuzz` (with double `zz`).

Reply:
@CristhianMotoche yes, alredy corrected.